main/test/ca/uwaterloo/flix/language/phase/TestNamer.scala

```diff
@@ -139,7 +127,66 @@ class TestNamer extends FunSuite with TestUtils {
          |}
        """.stripMargin
     val result = compile(input, DefaultOptions)
-    expectError[NameError.DuplicateDef](result)
+    expectError[NameError.DuplicateDefOrSig](result)
+  }
+
+  test("DuplicateDefOrSig.07") {
+    val input =
+      """
+        |class C[a] {
+        |    def f(x: a): Int
+        |    def f(x: a): Bool
+        |}
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.DuplicateDefOrSig](result)
+  }
+
+  test("DuplicateDefOrSig.08") {
+    val input =
+      """
+        |class C[a] {
+        |    def f(x: a): Int
+        |    def f(x: a): Bool
+        |    def f(x: Int): a
+        |}
+        |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.DuplicateDefOrSig](result)
+  }
+
+  test("DuplicateDefOrSig.09") {
+    val input =
+      s"""
+         |class A[a] {
+         |  def f(x: a): Int
+         |}
+         |
+         |namespace A {
+         |  def f(): Int = 21
+         |}
+       """.stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[NameError.DuplicateDefOrSig](result)
```

Comment:
I would like a few more test cases like this; with nested namespaces and classes.